illumos-utils/src/opte/port.rs

```diff
@@ -54,9 +36,10 @@ struct PortInner {
     vnic: String,
 }
 
+#[cfg(target_os = "illumos")]
```

Comment:
Has the door finally closed on non-illumos builds?  (I know I don't, but I was not sure if others do)

Reply:
It's still at the same level of support for non-illumos as previously, i.e. replacing illumos specific calls with stubs. But this reduces the duplication we had and also keeps a lot of the parameter validation and stuff consistent without having to hop between 2 files every time you want to change something in PortManager/Port.